tests/test_project.py

```diff
@@ -505,22 +505,22 @@ def op4(job):
         op3_ = project.operations['op3']
         op4_ = project.operations['op4']
         for job in project:
-            assert not op3_.eligible(job)
-            assert not op4_.eligible(job)
+            assert not op3_._eligible([job])
+            assert not op4_._eligible([job])
```

Comment:
This is not a pure refactoring PR, because this changes public API. I'd have preferred to have that separated out.

Reply:
This is a nice point. 
I can make a PR today which deprecates `eligible` and `complete` methods of `FlowGroup` and `BaseFlowOperation`

Reply:
If that's not too hard to do, then I think that would be good and we can simply rebase this branch here.